--- rpm-1.3.1/ext/rpm/rpm.c~	2013-04-28 03:28:24.000000000 +0300
+++ rpm-1.3.1/ext/rpm/rpm.c	2013-04-28 03:29:21.883894136 +0300
@@ -243,7 +243,7 @@
 #define DEFINE_DBI(name) \
 	rb_define_const(rpm_mRPM, "DBI_"#name, INT2NUM(RPMDBI_##name))
 	DEFINE_DBI(PACKAGES);
-	DEFINE_DBI(DEPENDS);
+//	DEFINE_DBI(DEPENDS);
 	DEFINE_DBI(LABEL);
 	DEFINE_DBI(ADDED);
 	DEFINE_DBI(REMOVED);
--- rpm-1.3.1/ext/rpm/package.c	2013-04-28 04:27:31.537204007 +0300
+++ rpm-1.3.1/ext/rpm/package.c	2013-04-28 04:23:55.247324566 +0300
@@ -683,7 +683,7 @@
 {
 	VALUE name;
 	const char* n;
-	headerNVR(RPM_HEADER(pkg), &n, NULL, NULL);
+	headerNEVRA(RPM_HEADER(pkg), &n, NULL, NULL, NULL, NULL);
 	name = n ? rb_str_new2(n) : Qnil;
 
 	return name;
@@ -735,7 +735,8 @@
 	const char* r;
 	VALUE e;
 
-	headerNVR(RPM_HEADER(pkg), NULL, &v, &r);
+	headerNEVRA(RPM_HEADER(pkg), NULL, NULL, &v, &r, NULL);
+
 	if (!v) {
 		ver = Qnil;
 	} else if (!r) {
