--- ruby-rpm-1.2.0/ext/rpm/extconf.rb~	2006-11-19 00:41:49.000000000 +1000
+++ ruby-rpm-1.2.0/ext/rpm/extconf.rb	2006-11-19 00:45:52.000000000 +1000
@@ -21,6 +21,15 @@
 
 def check_db
   dir_config('db')
+  if have_library("db-4.6","db_version")
+    return true
+  end
+  if have_library("db-4.5","db_version")
+    return true
+  end
+  if have_library("db-4.3","db_version")
+    return true
+  end
   if have_library("db-4.2","db_version")
     return true
   end
